src/components/Image/useIntersectionObserver.tsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+
+function useIntersectionObserver({
```

Comment:
this is the only part of this PR that works as expected (and only because I've copied it from that article)